code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -807,6 +815,18 @@ GLOBAL_LIST_EMPTY(PDAs)
 	else
 		to_chat(usr, "<span class='warning'>This PDA does not have a pen in it!</span>")
 
+/obj/item/pda/proc/eject_cart()
+	if(issilicon(usr) || !usr.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))

```

Comment:
Think telekenetics should apply here, no?

Reply:
TK doesn't work with the ID or pen and letting it work with the cart makes it teleport into your hand. I'm too lazy to add a check to make it drop like IDs in computers.